.github/workflows/e2e-suite.yml

```
@@ -16,28 +14,21 @@ jobs:
       - name: Update system packages
         run: sudo apt-get update -y
 
-      - name: Install system deps
-        run: sudo apt-get install -y build-essential bats parallel netcat
-
       - name: Setup Python
         uses: actions/setup-python@75f3110429a8c05be0e1bf360334e4cced2b63fa # pin@v2
```

```suggestion
        uses: actions/setup-python@v4
```
For official actions published by GitHub, let's pin version number instead of hash